From 8fe951a60cae896464585a1cfc6bbae2e825517f Mon Sep 17 00:00:00 2001
From: Andy <dahli.llama@gmail.com>
Date: Tue, 26 Nov 2024 09:35:07 -0600
Subject: [PATCH] fix: Issue #405 for General's Blade discounting all cards and
 not just units.

---
 CurrentEffectAbilities.php | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/CurrentEffectAbilities.php b/CurrentEffectAbilities.php
index a779546..7174eee 100644
--- a/CurrentEffectAbilities.php
+++ b/CurrentEffectAbilities.php
@@ -366,8 +366,10 @@ function CurrentEffectCostModifiers($cardID, $from, $reportMode=false)
           $remove = true;
           break;
         case "0414253215"://General's Blade
-          $costModifier -= 2;
-          $remove = true;
+          if($from != "PLAY" && DefinedTypesContains($cardID, "Unit", $currentPlayer)) {
+            $costModifier -= 2;
+            $remove = true;
+          }
           break;
         default: break;
       }
-- 
2.45.2.windows.1

